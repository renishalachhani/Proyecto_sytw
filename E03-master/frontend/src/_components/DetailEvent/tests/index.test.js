import jsdom from 'jsdom'
import React from 'react'
import renderer from 'react-test-renderer'
import { mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { DetailEvent } from '../index'

configure({ adapter: new Adapter() })

const { JSDOM } = jsdom
const dom = new JSDOM('<!doctype html><html><body></body></html>')
const { window } = dom

global.window = window
global.document = window.document

const defaultProps = {  
    title : "title",
    src: "/images/book.jpg",
    fecha: "8 de Septiembre",
    desc: "El autor estará firmando libros en la libreria Lemus de La Laguna" 
};

const renderComponent = (props) =>  
    <DetailEvent
        {...defaultProps}
        {...props}
    />;

describe('<DetailEvent />', () => {

  it('Comprobar que se renderiza DetailEvent', () => {
    const wrapper = mount(
        <DetailEvent
            title = "title"
            src = "/images/book.jpg"
            fecha = "8 de Septiembre"
            desc = "El autor estará firmando libros en la libreria Lemus de La Laguna"                
        />
    )
    expect(wrapper.find(DetailEvent)).toHaveLength(1)
  });

  it('Comprobar que los props contienen valores', () => {  
    const Component = mount(<DetailEvent {...defaultProps} />);
    expect((Component).prop('title')).toEqual('title');
    expect((Component).prop('src')).toEqual('/images/book.jpg');
    expect((Component).prop('fecha')).toEqual('8 de Septiembre');
    expect((Component).prop('desc')).toEqual('El autor estará firmando libros en la libreria Lemus de La Laguna');
  });

  it('Snapshot', () => {  
    const Component = renderer.create(<DetailEvent {...defaultProps} />).toJSON();
    expect(Component).toMatchSnapshot();
  });

})