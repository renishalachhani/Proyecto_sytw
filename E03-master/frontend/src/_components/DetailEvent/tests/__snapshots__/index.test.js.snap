// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailEvent /> Snapshot 1`] = `
<div>
  <img
    alt="Concierto"
    className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
    height="140"
    src="http://localhost:4000/images/book.jpg"
    title="title"
  />
  <div
    className="card-content"
  >
    <div
      className="media"
    >
      <div
        className="media-content"
      >
        <p
          className="title is-4"
        >
          title
        </p>
        <p
          className="subtitle is-6"
        >
          8 de Septiembre
        </p>
      </div>
    </div>
    <div
      className="content"
    >
      El autor estar√° firmando libros en la libreria Lemus de La Laguna
    </div>
  </div>
</div>
`;
